qa/tasks/cephfs/test_mds_metrics.py

```diff
@@ -511,3 +520,34 @@ def test_client_metrics_and_metadata(self):
             if not (client_metadata[i]['valid_metrics']):
                 raise RuntimeError("valid_metrics not found!")
 
+    def test_non_existing_mds_rank(self):
+        # validate
+        valid, metrics = self._get_metrics(
+            self.verify_mds_metrics(client_count=TestMDSMetrics.CLIENTS_REQUIRED), 30)
+        log.debug("metrics={0}".format(metrics))
+        self.assertTrue(valid)
+
+        filtered_mds = '15'
+        mds_metrics = metrics['metrics']
+        if f"mds.{filtered_mds}" in list(mds_metrics):
+            raise RuntimeError(f"filtered_mds={filtered_mds} already exists")
+        
+        # initiate a new query with `--mds_rank` filter
+        trials = 0
+        while(trials <= 5):
+            time.sleep(5)
+            metrics = json.loads(self.mgr_cluster.mon_manager.raw_cluster_cmd("fs", "perf", "stats", "--mds_rank", filtered_mds))
+            trials = trials + 1
```

Comment:
But calling `_get_metrics()` will make the test fail as we are passing a invalid mds_rank...which will try to get `metadata` and `global_metrics` for that which shouldn't be for our case.

Reply:
@neesingh-rh 
I meant something like this would be nice to have:
```
    def test_non_existing_mds_rank(self):
        def verify_filtered_metrics(metrics):
        #checks if the metrics has non empty client_metadata and global_metrics
            if metrics['client_metadata'] or metrics['global_metrics']:
                return True
            return False

        try:
            # validate
            filter_rank = 15
            valid, metrics = self._get_metrics(verify_filtered_metrics, 30,
                                               '--mds_rank={}'.format(filter_rank))
            log.info(f'metrics={metrics}')
            self.assertFalse(valid, "Fetched 'ceph fs perf stats' metrics using non existing mds rank")
        except MaxWhileTries:
            #success
            pass
```
If your patch exist `test_non_existing_mds_rank` would succeed and it would fail otherwise. 
Btw, I think `15` could be any number >=1 also, as the setUp() has only `_start_with_single_active_mds`. See that.

You could also create another test function that checks if we get only the filtered metrics from that mds.
I don't think 'test_query_mds_filter' does any specific checks. Or may be I would've miss something in test_query_mds_filter.

Reply:
Yup, I found that during testing that's why had to add a check for that also.I will add a test for that also.